Revamp appveyor.yml file.

djberg96 opened the pull request:
This is an update to the appveyor.yml file. A bit about the changes:

* Only tests the master branch. I'm not sure why we care about other branches but I can add them if you feel it's necessary.
* Uses Powershell in the "install" portion instead of cmd.exe. This makes it easier to skip gem installation if the dependencies we need are already installed. PS is just better in general.
* Removes all that Travis stuff. I have no idea what that's about. This ain't Travis.
* Caches the gem directories for each distribution. We don't need to re-provision gems every time we build, so this speeds up the build process.
* No after-test task. No idea why we care about that.

I tried it with local settings on my own fork and it worked fine. All the builds, including 1.9.3, were green.

segiddins commented on appveyor.yml:
```diff
@@ -1,38 +1,38 @@
----
-version: "{build}"
+version: '{build}'
 branches:
```
We do want these other branches

djberg96 replied:
@segiddins Why, do we ever backport? It's no problem to add them, just curious.

segiddins replied:
Yes, we have in the past and might need to in the future, especially for security-related issues.

simi commented:
What about this [line](https://github.com/rubygems/rubygems/blob/d0efa5e534f0c56bbc59130e1ccb453fb43a57c5/test/rubygems/test_bundled_ca.rb#L11)? Is that `ENV['TRAVIS']` really Travis specific or CI specific? I think both Travis and Appveyor exports `ENV['CI']`.

djberg96 commented:
@segiddins Ok, I've updated the PR to include the additional branches.

@simi I've updated the PR so that it sets ENV['TRAVIS'] first. You're right that appveyor does export ENV["CI"] - it's set to true - so we should probably look at converting all instances of those in a future PR if "CI" serves as a sort of universal environment variable for continuous integration services in general.

segiddins commented:
@duckinator I'll let you review this

duckinator commented:
@simi we export the `TRAVIS` environment variable ourself. I'll look into this part as a separate PR after I  catch up on my backlog -- I'll either use one they both already export, or renaming the one we export to be more generic. :)

@segiddins looking at it now.

duckinator commented:
@djberg96 this looks good to me! You even got 1.9.3 tests working on Appveyor. :D

@homu r+

homu commented:
:pushpin: Commit e66fd6f has been approved by `duckinator`

<!-- @homu r=duckinator e66fd6f26d73dbc75f3e1d5aae657aeb8eb0aca5 -->

homu commented:
:zap: Test exempted - [status](https://travis-ci.org/rubygems/rubygems/builds/125059939)
